gen/utils.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+type StringListFlag []string
+
+var StringListFlagI flag.Value = &StringListFlag{}
+
+func (f *StringListFlag) String() string     { return fmt.Sprintf("%#v", f) }
+func (f *StringListFlag) Set(v string) error { *f = append(*f, v); return nil }
+
+type CommentType uint8
+
+const (
+	CommentTypeSingleLine CommentType = iota
+	CommentTypeMultiLine
+)
+
+func Comment(in string, mode CommentType, indent int, newline bool) string {
+	if in == "" || strings.TrimSpace(in) == "TODO" {
+		return ""
+	}
+
+	const space = ' '
+	var out strings.Builder
+	if newline {
+		out.WriteString("\n")
+	}
+	if mode == CommentTypeMultiLine {
+		for i := 0; i < indent; i++ {
+			out.WriteRune(space)
+		}
+		out.WriteString("/**\n")
+	}
+	sc := bufio.NewScanner(strings.NewReader(strings.TrimSpace(in)))
+	for sc.Scan() {
+		line := sc.Text()
+		for i := 0; i < indent; i++ {
+			out.WriteRune(space)
+		}
+		switch mode {
+		case CommentTypeSingleLine:
+			out.WriteString("// ")
+		case CommentTypeMultiLine:
+			out.WriteString(" * ")
+		default:
+			panic("unreachable")
+		}
+		out.WriteString(line)
+		out.WriteString("\n")
+	}
+	if mode == CommentTypeMultiLine {
+		for i := 0; i < indent; i++ {
+			out.WriteRune(space)
+		}
+		out.WriteString(" */")
+	}
+	return out.String()
+}
+
+func ConstantCase(v string) string {
+	return strings.ToUpper(v)
+}
+
+func PascalCase(s string) string {
+	var out strings.Builder
+	out.Grow(len(s))
+	nextUpper := true
+	for i := 0; i < len(s); i++ {
```

Comment:
```suggestion
	for _, c := range s {
```

Which will correctly decode the runes from the string. You can then remove the `rune(c)` from below.